homeassistant/components/kef/media_player.py

```diff
@@ -0,0 +1,269 @@
+"""Platform for the KEF Wireless Speakers."""
+
+import datetime
+import logging
+
+import voluptuous as vol
+from aiokef.aiokef import AsyncKefSpeaker
+from homeassistant.components.media_player import (
+    PLATFORM_SCHEMA,
+    SUPPORT_SELECT_SOURCE,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_STEP,
+    MediaPlayerDevice,
+)
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_TYPE,
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNKNOWN,
+)
+from homeassistant.helpers import config_validation as cv
+
+_CONFIGURING = {}
+_LOGGER = logging.getLogger(__name__)
+
+
+DEFAULT_NAME = "KEF"
+DEFAULT_PORT = 50001
+DEFAULT_MAX_VOLUME = 0.5
+DEFAULT_VOLUME_STEP = 0.05
+DEFAULT_INVERSE_SPEAKER_MODE = False
+
+DOMAIN = "kef"
+
+SCAN_INTERVAL = datetime.timedelta(seconds=30)
+PARALLEL_UPDATES = 0
+
+SOURCES = {"LSX": ["Wifi", "Bluetooth", "Aux", "Opt"]}
+SOURCES["LS50"] = SOURCES["LSX"] + ["Usb"]
+
+SUPPORT_KEF = (
+    SUPPORT_VOLUME_SET
+    | SUPPORT_VOLUME_STEP
+    | SUPPORT_VOLUME_MUTE
+    | SUPPORT_SELECT_SOURCE
+    | SUPPORT_TURN_OFF
+    | SUPPORT_TURN_ON
+)
+
+CONF_MAX_VOLUME = "maximum_volume"
+CONF_VOLUME_STEP = "volume_step"
+CONF_INVERSE_SPEAKER_MODE = "inverse_speaker_mode"
+CONF_STANDBY_TIME = "standby_time"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_TYPE): vol.In(["LS50", "LSX"]),
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MAX_VOLUME, default=DEFAULT_MAX_VOLUME): cv.small_float,
+        vol.Optional(CONF_VOLUME_STEP, default=DEFAULT_VOLUME_STEP): cv.small_float,
+        vol.Optional(
+            CONF_INVERSE_SPEAKER_MODE, default=DEFAULT_INVERSE_SPEAKER_MODE
+        ): cv.boolean,
+        vol.Optional(CONF_STANDBY_TIME): vol.In([20, 60]),
```

Comment:
Is this the correct way of doing this?

This variable can also be `None`. Should I use `vol.In([20, 60, None])` instead?

Reply:
Since only these three values are valid I would say you should include it in the list.
However, I would specify a `DEFAULT_STANDBY_TIME` which has the value `None` and set it as the default value.
So this would be the result:
```python
DEFAULT_STANDBY_TIME = None
...
vol.Optional(CONF_STANDBY_TIME, default=DEFAULT_STANDBY_TIME): vol.In([None, 20, 60])
```
Still I would think about changing this default value to something else e.g. `0` or `false`.

Reply:
Thanks for your comment, at the Discord chat, @balloob actually recommended me to not define a `DEFAULT_STANDBY_TIME`.

Reply:
That is even better!

Reply:
I mean the variable `DEFAULT_STANDBY_TIME = None` because the absence of it makes it `None` automatically because of the use of `get`.

Reply:
Right, `get` returns `None` if it does not find anything. So is there anything left to do here? :)

Reply:
Nothing really. Just waiting on a review from core. 😄

Reply:
Oh I see 😄 well you adressed all requested changes, so this should be a quick review from the core team.